src/equations/equations.jl

```diff
@@ -274,13 +274,14 @@ terms are present.
 function energy_total_modified(q_global, equations::AbstractShallowWaterEquations, cache)
     # `q_global` is an `ArrayPartition` of the primitive variables at all nodes
     @assert nvariables(equations) == length(q_global.x)
+    # tmp1 is always in the cache
+    @unpack tmp1 = cache
```

Comment:
I am not sure, I fully understand. So `get_tmp` would just be `get_tmp(cache) = cache.tmp1` to deliver a temporary storage for one field? Where exactly do we need a temporary storage for a full `ArrayPartition`? The `@unpack tmp1 = semi.cache` in the relaxation callback is also just for one field.

Reply:
We may use `get_tmp(equations, cache)` or even including the `solver`, too, to have more options for dispatch. It can be kind of annoying to have to fix one variable name in the `cache`. But it works for now, so we don't have to change it right now.

The full vector cache could be used in 
https://github.com/JoshuaLampert/DispersiveShallowWater.jl/blob/cf20e3f45485714167f85d9f5c1fc70ed1dd3b1f/src/callbacks_step/relaxation.jl#L84

Reply:
Ok, let's keep this for a future PR. I think this one is already big enough.

> The full vector cache could be used in

Ah yes, I forgot about that. That  this is allocating is also already tracked in #11. Thanks!